expression/builtin_cast_vec.go

```diff
@@ -776,11 +776,56 @@ func (b *builtinCastStringAsTimeSig) vecEvalTime(input *chunk.Chunk, result *chu
 }
 
 func (b *builtinCastDecimalAsIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDecimalAsIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDecimal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	d64s := buf.Decimals()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+
+		var to types.MyDecimal
+		err = d64s[i].Round(&to, 0, types.ModeHalfEven)
+		if err != nil {
+			result.SetNull(i, true)
+			continue
+		}
+
+		if !mysql.HasUnsignedFlag(b.tp.Flag) {
+			i64s[i], err = to.ToInt()
+		} else if b.inUnion && to.IsNegative() {
+			i64s[i] = 0
+		} else {
+			var uintRes uint64
+			uintRes, err = to.ToUint()
+			i64s[i] = int64(uintRes)
+		}
+
+		if types.ErrOverflow.Equal(err) {
+			warnErr := types.ErrTruncatedWrongVal.GenWithStackByArgs("DECIMAL", d64s[i])
+			err = b.ctx.GetSessionVars().StmtCtx.HandleOverflow(err, warnErr)
+		}
+
+		if err != nil {
+			return err
+		}
```

Comment:
Can we set `to = d64s[i]` to avoid defining a new MyDecimal every time? And line 802 can be moved to line 796.